File: boj/b_1874.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

n = int(input())
result, s, a = [], [], 1
for _ in range(n):
    now = int(input())
    if s and s[-1] == now:
        s.pop()
        result.append("-")
    else:
        for i in range(a, now + 1):
            s.append(i)
            result.append("+")
        s.pop()
        result.append("-")
        a = now + 1
if not s:
    for i in result:
        print(i)
else:
    print("NO")

File: boj/b_1874.js
function solution(arr) {
  let [result, s, a] = [[], [], 1];
  arr.forEach((v) => {
    if (s.length && s[s.length - 1] == v) {
      s.pop();
      result.push("-");
    } else {
      for (let i = a; i <= v; i += 1) {
        s.push(i);
        result.push("+");
      }
      s.pop();
      result.push("-");
      a = v + 1;
    }
  });
  if (!s.length) return result;
  else return "NO";
}
console.log(solution([4, 3, 6, 8, 7, 5, 2, 1]));
